### PR TITLE
Add board details and reporting to documentation utilities

### PR DESCRIPTION
**Pull request by timchinowsky**
- Enhances `docs/shared_bindings_matrix.py` to optionally add more board details to the returned dict
- Adds `docs/board_summaries.py`, which creates `.csv` reports from that dict, enabling visualizations like [this](https://docs.google.com/spreadsheets/d/1fZiNTo8ByxCGwQAtYSaaCKXIRkBdApn435xuSkT-nzA/edit?usp=sharing)
- Useful for answering questions like "what boards use which ports and processors?",  "what boards have the most pins?", and "for what ports/processors is this module implemented?"
- Not sure if y'all want it in the repo, but I've found it helpful, thought I'd share

### CODE REVIEW

File: docs/board_summaries.py
Comment (tannewt) [inline docs/board_summaries.py:1]: Since this isn't used in doc generation, mind moving it to tools?

**Review by tannewt — CHANGES_REQUESTED** [review]
One request. Looks good otherwise. Neat spreadsheet!

File: docs/board_summaries.py
Comment (timchinowsky) [reply docs/board_summaries.py:1]: Wonder if you've got a prettier way of doing this relative import:

https://github.com/timchinowsky/circuitpython/blob/a3d7a2c4df70b69604fb4fb99d664155e3c70a2f/tools/board_summaries.py#L27C2-L27C3

Comment (tannewt) [reply docs/board_summaries.py:1]: Nah, this we do this already: https://github.com/adafruit/circuitpython/blob/d8ca8421a09b1e5fee8fd29213c3ccc7914f5d8c/tools/build_release_files.py#L16

**Review by tannewt — APPROVED** [review]
Thank you!

**Pull request merged**